src/components/dropdownSearch/MultiSelectDropdownSearch/MultiSelectDropdownSearchReducer.tsx

```diff
@@ -0,0 +1,134 @@
+import { DropdownSearchActions, IOptionsDropdownSearchPayload } from '../DropdownSearchActions';
+import {
+  addUniqueSelectedOption,
+  dropdownSearchInitialState,
+  dropdownSearchReducer,
+  getDisplayedOptions,
+  getNextIndexPosition,
+  getOptionsFiltered,
+  IDropdownSearchState,
+  removeSelectedOption,
+} from '../DropdownSearchReducers';
+import { IReduxAction } from '../../../utils/ReduxUtils';
+import {FixedQueue} from '../../../utils/FixedQueue';
+import {IDropdownOption} from '../DropdownSearch';
+import {keyCode} from '../../../utils/InputUtils';
+
+export const multiSelectDropdownSearchReducer = (state: IDropdownSearchState = dropdownSearchInitialState,
+  action: IReduxAction<IOptionsDropdownSearchPayload>) => {
+
+  switch (action.type) {
+    case DropdownSearchActions.addMultiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        options: action.payload.optionsDropdown,
+        selectedOptions: new FixedQueue<IDropdownOption>(),
+        filterText: action.payload.filterText,
+        isOpened: false,
+      };
+    case DropdownSearchActions.removeSelectedOption:
+      state.selectedOptions = removeSelectedOption(state, action.payload.selectedOptionValue);
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
```

Comment:
done but with a twist maybe you could re-check it, but I create a new array in with util methods